daq/runner.py

```diff
@@ -985,18 +998,15 @@ def _target_set_finalize(self, device, result_set, reason):
         suppress_tests = self.fail_mode or self.result_linger
         if results and suppress_tests:
             LOGGER.warning('Suppressing further tests due to failure.')
-            self.run_tests = False
+            self._run_tests = False
             if self.result_linger:
                 self._linger_exit = 1
         self._result_sets[device] = result_set
 
-        if self.run_limit and self.run_count >= self.run_limit and self.run_tests:
+        if self._run_limit and self._run_count >= self._run_limit and self._run_tests:
             LOGGER.warning('Suppressing future tests because run limit reached.')
-            self.run_tests = False
-        if self.single_shot and self.run_tests:
-            LOGGER.warning('Suppressing future tests because test done in single shot.')
-            self._handle_faucet_events()  # Process remaining queued faucet events
```

Comment:
I think this last faucet event processing can be left in.

Reply:
But _should_ it be left in (and if so, why)?  I was trying to figure out what that code was for and it seemed out of place, so I was trying to clean things up a bit.

Reply:
The last faucet events check should only go under single shot mode so that line needs to be moved / be under another if statement. 
The reasoning for that line is it is possible for a device to finish testing before any other devices' learning events are processed considering during DAQ startup, there is a flood of events and many events are also debounced. To minimize that problem, I added in this last event processing to trigger any remaining devices to make integration tests more stable.